### PR TITLE
Update XML readme

### PR DESCRIPTION
**Pull request by mristin**
We document how we constructed the schema and its background.

Additionally, we clarify a bit on roundtrips XML -> JSON -> XML.

### CODE REVIEW

File: schemas/xml/README.md [hunk 1]
```diff
@@ -1,57 +1,449 @@
...
+<someAttribute>
+    <anotherAttribute>
+        <!-- ... -->
+    </anotherAttribute>
+    <yetAnotherAttribute>
+        <!-- ... -->
+    </yetAnotherAttribute>
+</someAttribute>
+```
+
```
Comment (BirgitBoss) [inline schemas/xml/README.md:196]: This means we are not backward compatible if we would introduce inheritance in a later version? Introducing inheritance on UML level would be backward compatible.

File: schemas/xml/README.md [hunk 2]
```diff
@@ -1,57 +1,449 @@
...
+The type information about `AnotherClass` is omitted, as the type of the `SomeClass/someAttribute` is fixed in the meta-model.
+
+### Attribute Values as Aggregations
+
+Many attributes in the meta-model do not represent a single value (be it primitive or structured as a class), but aggregate instances of meta-model classes.
+For example, [Submodel/submodelElement] aggregates instances of [SubmodelElement]'s.
+
+If we just concatenated all the properties of the instances, we would not which property belongs to which instance (or that would be complicated).
```
Comment (BirgitBoss) [inline schemas/xml/README.md:204]: editorial: we would not know - know is missing

**Review by BirgitBoss — COMMENTED** [review]
same findings as for JSON

File: schemas/json/README.md [hunk 3]
```diff
@@ -51,6 +54,10 @@ Attributes describing aggregations, *i.e.*, the attributes with the cardinality
 
 [JSON arrays]: https://json-schema.org/understanding-json-schema/reference/array.html
 
+We explicitly forbid empty JSON arrays to avoid confusion about attributes which have cardinality ``0..*``.
+Namely, an empty array is semantically equal to an omitted attribute (according to the meta-model).
+Thus, the JSON property representing an aggregation attribute must be omitted if the aggregation is empty.
+
 For better readability, we diverge from the meta-model and spell most properties in plural.
```
Comment (BirgitBoss) [inline schemas/json/README.md:61]: not for better readability, see my original comment

**Review by BirgitBoss — COMMENTED** [review]
for JSON: Cardinality 0..* still open
for XML: very similar findings to JSON
Thanks for detailled explanation with examples for Discriminators

File: schemas/json/README.md [hunk 4]
```diff
@@ -51,6 +54,10 @@ Attributes describing aggregations, *i.e.*, the attributes with the cardinality
...
 For example, ``submodelElements`` instead of ``submodelElement`` in case of [Submodel] class.
```
[the remark this answers is not in the document]
Comment (mristin) [reply schemas/json/README.md:62]: Done.

File: schemas/xml/README.md [hunk 5]
```diff
@@ -1,57 +1,449 @@
...
-	<aas:dataSpecifications></aas:dataSpecifications>
-	<aas:submodels></aas:submodels>
-</aas:aasenv>
-```  
-  
-Further examples are in the [example folder](examples)
+# XML
+
+Extensible Markup Language ([XML]) is a popular serialization format for data exchange and storage.
+Admittedly, XML is not as widely used for client-server communication compared to other formats such as JSON.
```
[the remark this answers is not in the document]
Comment (mristin) [reply schemas/xml/README.md:4]: Removed.

File: schemas/xml/README.md [hunk 6]
```diff
@@ -1,57 +1,449 @@
...
+Nevertheless, XML is still leveraged in many databases and as a file format within a rich ecosystem of accompanying validation, query and transformation tools.
```
[the remark this answers is not in the document]
Comment (mristin) [reply schemas/xml/README.md:5]: Removed.

File: schemas/xml/README.md [hunk 7]
```diff
@@ -1,57 +1,449 @@
...
+
+[Background]: #Background
+
+We finally decided that the drawbacks outlined above outweighed the advantage in form of succinct representation.
+
+### Cardinality
+
+If an attribute has cardinality ``0..1`` or ``0..*`` and is not specified in the instance, the element is simply omitted.
+
+For aggregation attributes, *i.e.* with cardinality upper bound higher than 1 such as ``0..*``, we require the respective XML elements to be given in plural form for better readability and diverge thus from the meta-model.
```
[the remark this answers is not in the document]
Comment (mristin) [reply schemas/xml/README.md:85]: Done.

File: schemas/xml/README.md [hunk 8]
```diff
@@ -1,57 +1,449 @@
...
+For example, ``submodelElements`` instead of ``submodelElement`` in case of [Submodel] class.
+
+In cases where plural form made no sense for an attribute, we kept it as-is in singular form (*e.g.*, `isCaseOf`).
```
[the remark this answers is not in the document]
Comment (mristin) [reply schemas/xml/README.md:88]: Done.

File: schemas/xml/README.md [hunk 9]
```diff
@@ -1,57 +1,449 @@
...
+
+### Attribute Values of Simple Data Types
+
+The attribute values given in the meta-model as simple data types (see Section [5.7.12 Primitive and Simple Data Types] of the meta-model) are serialized as text enclosed in the XML element corresponding to the attribute.
```
[the remark this answers is not in the document]
Comment (mristin) [reply schemas/xml/README.md:92]: Referred to Section why-no-xml-attributes.

File: schemas/xml/README.md (hunk 2 above)
Comment (mristin) [reply schemas/xml/README.md:204]: Fixed.

File: schemas/json/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (mristin) [reply schemas/json/README.md:61]: Re-phrased.

**Pull request merged**